rbac/management/group/definer.py

```diff
@@ -167,8 +167,9 @@ def add_roles(group, roles_or_role_ids, tenant, user=None):
 
         system_policy.roles.add(role)
 
-        dual_write_handler = RelationApiDualWriteGroupHandler(group, ReplicationEventType.ASSIGN_ROLE, [])
-        dual_write_handler.replicate_added_role(role)
+        if group.tenant.tenant_name != "public":
```

Comment:
Oh, I just realize that we can make use of the system flag in this case. The system True group are all in the public tenant, then we can save a query to get the tenant

Reply:
Sometimes tests use system groups outside the public tenant so I'm worried relying on that implicit rule is a little fragile. We want to check for the public tenant specifically, and this does that directly.

That said, both methods are already passed the tenant so we can just use that. Thanks for pointing out the extra query.

Reply:
Hrm, although I'm not sure why the method queries for the group when it's already given one. But, changing that is probably more invasive than we need right now. Because it does that though (the extra query above with the `tenant`), it's looking like using the `tenant` variable is viable at least. WDYT?

Reply:
👍 Ah ha, the tenant is already there, lol, nice!